src/codemodder/context.py

```diff
@@ -85,28 +86,41 @@ def get_failed_files(self):
             )
         )
 
-    def process_dependencies(self, codemod_id: str):
+    def process_dependencies(
+        self, codemod_id: str
+    ) -> dict[Dependency, PackageStore | None]:
         """Write the dependencies a codemod added to the appropriate dependency
-        file in the project.
+        file in the project. Returns a dict listing the locations the dependencies were added.
         """
         dependencies = self.dependencies.get(codemod_id)
         if not dependencies:
-            return
+            return {}
 
-        dependencies_store = self.repo_manager.dependencies_store
-        if dependencies_store is None:
+        # populate everything with None and then change the ones added
+        record: dict[Dependency, PackageStore | None] = {}
+        for dep in dependencies:
+            record[dep] = None
+
+        store_list = self.repo_manager.package_stores
+        if not store_list:
             logger.info(
                 "unable to write dependencies for %s: no dependency file found",
                 codemod_id,
             )
-            return
+            return record
 
         # pylint: disable-next=cyclic-import
         from codemodder.dependency_management import DependencyManager
 
-        dm = DependencyManager(dependencies_store, self.directory)
-        if (changeset := dm.write(list(dependencies), self.dry_run)) is not None:
-            self.add_results(codemod_id, [changeset])
+        for package_store in store_list:
+            dm = DependencyManager(package_store, self.directory)
+            if (changeset := dm.write(list(dependencies), self.dry_run)) is not None:
+                self.add_results(codemod_id, [changeset])
```

Comment:
Maybe a debug log here saying which file is being written?

Reply:
This information is being store in the `record` variable. The debug message with the file the dependency has been written in is in the new `record_dependency_update` function at `codemodder.py`